packages/plugin-inline/test/Inline.test.ts

```diff
@@ -0,0 +1,161 @@
+import { describePlugin } from '../../utils/src/testUtils';
+import { Inline, FormatParams } from '../Inline';
+import JWEditor from '../../core/src/JWEditor';
+import { Constructor } from '../../utils/src/utils';
+import { Format } from '../Format';
+import { BasicEditor } from '../../../bundles/BasicEditor';
+import { InsertTextParams } from '../../plugin-char/Char';
+import { BoldFormat } from '../../plugin-bold/BoldFormat';
+import { UnderlineFormat } from '../../plugin-underline/UnderlineFormat';
+
+const insertText = async function(editor, text: string): Promise<void> {
+    const params: InsertTextParams = {
+        text: text,
+    };
+    await editor.execCommand('insertText', params);
+};
+const toggleFormat = async (editor: JWEditor, FormatClass: Constructor<Format>): Promise<void> => {
+    const formatParams: FormatParams = {
+        FormatClass: FormatClass,
+    };
+    await editor.execCommand('toggleFormat', formatParams);
+};
+
+describePlugin(Inline, testEditor => {
+    describe('InlineFormatDomParser', () => {
+        it('should parse text with nested similar formats', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore:
+                    '<p>a<span class="outer">bc<span class="inner">de</span>fg</span>h</p>',
+                contentAfter:
+                    '<p>a<span class="outer">bc</span><span class="outer"><span class="inner">de</span></span><span class="outer">fg</span>h</p>',
+            });
+        });
+        it('should parse some nested formats in order', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>ab<b>cd<i>ef<u>gh</u>ij</i>kl</b>mn</p>',
+                contentAfter:
+                    '<p>ab<b>cd</b><b><i>ef</i></b><b><i><u>gh</u></i></b><b><i>ij</i></b><b>kl</b>mn</p>',
+            });
+        });
+    });
+    describe('toggleFormat', () => {
+        describe('Selection collapsed', () => {
+            it('should make bold the next insertion', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '[]a',
+                    stepFunction: async (editor: JWEditor) => {
+                        await toggleFormat(editor, BoldFormat);
+                        await insertText(editor, 'b');
+                    },
+                    contentAfter: '<b>b[]</b>a',
+                });
+            });
+            it('should not make bold the next insertion when toggleFormat 2 times', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '[]a',
+                    stepFunction: async (editor: JWEditor) => {
+                        await toggleFormat(editor, BoldFormat);
+                        await toggleFormat(editor, BoldFormat);
+                        await insertText(editor, 'b');
```

Comment:
I don't think moving these tests made sense. Now you're calling `insertText` in `Inline` which does not depend on `Char`, while previously you were calling `toggleFormat` in `Char`, which *does* depend on `Inline` so it was fine, now it's not.